### ISSUE TITLE
TS compiler to throw hands in the air about no explicit returns when {} are used in the lambda

### ISSUE THREAD

**Issue by pmonty** [Microsoft/TypeScript#19514]
TS compiler to throw hands in the air about no explicit returns when {} are used in the lambda

**TypeScript Version:**  2.3.5

**Code**

```ts
const filtered = this.state.emailAddressesViewList.findIndex((email: EmailAddressViewModel) => {
    (email.EmailAddress === emailAddress);
});
```
**Expected behavior:**

Want to return the index of the element in the array

**Actual behavior:**

It was always returning -1

**Solution**

Worked with either of these though, just would be nice to throw an error of some sort if possible

```
const filtered = this.state.emailAddressesViewList.findIndex((email: EmailAddressViewModel) => 
     (email.EmailAddress === emailAddress)
);
```

```
const filtered = this.state.emailAddressesViewList.findIndex((email: EmailAddressViewModel) => {
     return (email.EmailAddress === emailAddress)
});
```

**Comment by mhegazy** [comment]
Should be flagged as error under `--noImplicitReturns`.

**Comment by andy-ms** [comment]
@pmonty Could you share a complete example? I correctly get an error using `[].findIndex`, so I'm wondering if `this.state.emailAddressesViewList` is not a regular array.

**Comment by pmonty** [comment]
this.state.emailAddressesViewList is an Array of EmailAddressViewModel which is just

```
    export class EmailAddressViewModel {
        DisplayName: string;
        EmailAddress: string;
        isList: boolean;

        constructor(name: string, emailAddress: string, isList: boolean) {
            this.DisplayName = name;
            this.EmailAddress = emailAddress;
            this.isList = isList;
        }

        public getFriendlyLabel(): string {
            return this.DisplayName === "" ? this.EmailAddress : this.DisplayName;
        }
    }
```
```
        private isEmailInEmailList(emailAddress: string): number {
            const filtered = this.state.emailAddressesViewList.findIndex((email: EmailAddressViewModel) => {
               (email.EmailAddress === emailAddress)
            });

            return filtered;
        }
```

**Comment by andy-ms** [comment]
@pmonty When I try that out I see an error:
```ts
export class EmailAddressViewModel {
	DisplayName: string;
	EmailAddress: string;
	isList: boolean;

	constructor(name: string, emailAddress: string, isList: boolean) {
		this.DisplayName = name;
		this.EmailAddress = emailAddress;
		this.isList = isList;
	}

	public getFriendlyLabel(): string {
		return this.DisplayName === "" ? this.EmailAddress : this.DisplayName;
	}
}

class Foo {
	state: { emailAddressesViewList: EmailAddressViewModel[] }

	private isEmailInEmailList(emailAddress: string): number {
		const filtered = this.state.emailAddressesViewList.findIndex((email: EmailAddressViewModel) => {
		   (email.EmailAddress === emailAddress)
		});

		return filtered;
	}
}
```

```
src/a.ts(21,64): error TS2345: Argument of type '(email: EmailAddressViewModel) => void' is not assignable to parameter of type '(value: EmailAddressViewModel, index: number, obj: EmailAddressViewModel[]) => boolean'.
  Type 'void' is not assignable to type 'boolean'.
```
